### DISCUSSION TITLE
concurrent connection

### DISCUSSION THREAD

**Discussion by storynana** [capacitor-community/sqlite#436] [Q&A] [answered]
Hi, there;
     I am using this in one of my ionic projects. Is there any way I can use this CapacitorSQLite module to have concurrent access to my sqlite db? I am getting error when trying open two connections at the same time:
'CreateConnection: Connection xxx already exists.

Cheers,
Rob

**Comment by jepiqueau** [comment]
@storynana you can have only one connection in write access mode and one connection in read-only mode to the same database and you can have multiple connections to several databases.

**Comment by storynana** [comment]
Thanks Jean for the quick response.

Can I keep the connection always open during the lifetime of the app? Or it
should be a good practice to always close the connection when I completed
the query?

Cheers,
Rob.




On Sun, Jul 2, 2023 at 4:50 PM QUEAU Jean Pierre ***@***.***>
wrote:

> @storynana <https://github.com/storynana> you can have only one
> connection in write access mode and one connection in read-only mode to the
> same database and you can have multiple connections to several databases.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/capacitor-community/sqlite/discussions/436#discussioncomment-6335092>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACQQ76X5B5RJ6YZJREHSR6TXOD4Z3ANCNFSM6AAAAAAZ3GQTBU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Cheers,
qiuwei

**Comment by jepiqueau** [comment, marked as answer]
@storynana it is up to you but it is time consuming to open and close it each time for à query. I use to leave it open
